gtk4/src/subclass/cell_layout.rs

```diff
@@ -0,0 +1,489 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::subclass::prelude::*;
+use crate::{CellArea, CellLayout, CellRenderer, TreeIter, TreeModel};
+use glib::translate::*;
+use glib::{Cast, IsA, ObjectExt, Quark};
+use once_cell::sync::Lazy;
+
+#[derive(Debug)]
+pub struct CellLayoutDataCallback {
+    callback: ffi::GtkCellLayoutDataFunc,
+    user_data: glib::ffi::gpointer,
+    destroy_notify: glib::ffi::GDestroyNotify,
+}
+
+impl CellLayoutDataCallback {
+    pub fn call<C: IsA<CellLayout>, R: IsA<CellRenderer>, M: IsA<TreeModel>>(
+        &self,
+        cell_layout: &C,
+        cell_renderer: &R,
+        model: &M,
+        iter: &TreeIter,
+    ) {
+        unsafe {
+            if let Some(callback) = self.callback {
+                callback(
+                    cell_layout.as_ref().to_glib_none().0,
+                    cell_renderer.as_ref().to_glib_none().0,
+                    model.as_ref().to_glib_none().0,
+                    mut_override(iter.to_glib_none().0),
+                    self.user_data,
+                );
+            }
+        }
+    }
+}
+
+impl Drop for CellLayoutDataCallback {
+    fn drop(&mut self) {
+        unsafe {
+            if let Some(destroy_notify) = self.destroy_notify {
+                destroy_notify(self.user_data);
+            }
+        }
+    }
+}
+
+pub trait CellLayoutImpl: ObjectImpl {
+    fn add_attribute<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        attribute: &str,
+        column: i32,
+    ) {
+        self.parent_add_attribute(cell_layout, cell, attribute, column)
+    }
+
+    fn clear_attributes<R: IsA<CellRenderer>>(&self, cell_layout: &Self::Type, cell: &R) {
+        self.parent_clear_attributes(cell_layout, cell)
+    }
+
+    fn cells(&self, cell_layout: &Self::Type) -> Vec<CellRenderer> {
+        self.parent_cells(cell_layout)
+    }
+
+    fn set_cell_data_func<R: IsA<CellRenderer>>(
+        &self,
+        cell_layout: &Self::Type,
+        cell: &R,
+        callback: Option<&CellLayoutDataCallback>,
```

Comment:
```suggestion
        callback: Option<&CellLayoutDataCallback>,
```

No reference here and in the `parent` function. Somehow the callback needs to be stored by the impl.